backend/api_calls.py

```diff
@@ -794,6 +794,131 @@ def AutoUpdatePlayerGroups(firebase, player_id, new_player=False):
       AddPlayerGroupMappings(firebase, group_id, player_id)
 
 
+# TODO Decide how to make a life code as used up.
+def Infect(request, firebase):
+  """Infect a player via life code.
+
+  Infect a human and get 100 points.
+
+  Args:
+    playerId: The person doing the infecting.
+    lifeCodeId: The life code being taken/infected, makes to the victom.
+
+
+  Validation:
+    Valid IDs. Infector can infect or is self-infecting. Infectee is human.
+
+  Firebase entries:
+    /games/%(gameId)/players/%(playerId)
+    /groups/%(groupId) indirectly
+  """
+  valid_args = ['playerId', 'lifeCodeId']
+  required_args = list(valid_args)
+  helpers.ValidateInputs(request, firebase, required_args, valid_args)
+
+  player_id = request['playerId']
+  life_code_id = request['lifeCodeId']
+  victim_id = helpers.LifeCodeToPlayer(firebase, request['lifeCodeId'])
+  game_id = helpers.PlayerToGame(firebase, player_id)
+
+  player = {
+    'allegiance': firebase.get('/games/%s/players/%s' % (game_id, player_id), 'allegiance'),
+    'can_infect': firebase.get('/players/%s' % player_id, 'canInfect')
+  }
+  victim = {
+    'allegiance': firebase.get('/games/%s/players/%s' % (game_id, victim_id), 'allegiance'),
+    'can_infect': firebase.get('/players/%s' % victim_id, 'canInfect')
+  }
+
+  # Both players must be in the same game.
+  if helpers.PlayerToGame(firebase, victim_id) != game_id:
```

Comment:
Sorry. Check 2 uses player[canInfect] and check 3 uses victim[allegiance] so this can't be reordered much. The first check can move but that would break them up.